plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs

```diff
@@ -626,7 +705,31 @@ enterComputeCek costs = computeCek where
       case resultOrErr of
           -- Turn the cause of a possible failure, being a 'CekValue', into a 'Term'.
           Left e       -> throwCek $ mapCauseInMachineException (void . dischargeCekValue) e
-          Right result -> returnCek ctx result
+          Right result -> returnCek unbudgetedSteps ctx result
+
+    -- | Spend the budget that has been accumulated for a number of machine steps.
+    spendAccumulatedBudget :: WordArray -> CekM s ()
+    spendAccumulatedBudget !unbudgetedSteps = iforWordArray unbudgetedSteps spend
```

Comment:
I think we should at least change this and instead of requiring the caller to provide a `spend` function, require it to provide a `spendAccumulatedBudget` function, which would allow us in the non-tallying case not to iterate through the array at all.

Reply:
> instead of requiring the caller to provide a `spend` function

* `spendBudgetCek` (the edit button stopped working for some reason).

>  not to iterate through the array at all

... and just directly spend from the `7`th field that we calculate anyway (we do assume that all kinds of steps cost the same, right?).

Although we perhaps still need `spendBudgetCek` to deal with builtins? Well then perhaps we can have both.

Reply:
We *don't* assume that they all cost the same, currently. If we did, then yeah, we could spend directly from the 7th field. But e.g. I think Kenneth needs them to cost different amounts. So I was trying to keep that.

Reply:
> We don't assume that they all cost the same, currently.

Ok, then I misunderstood something, ignore me.